### PR TITLE
Added Mail must end with @devmob.com

### PR DESCRIPTION
**Pull request by okapifan**

### CODE REVIEW

**Review by mbicknese — COMMENTED** [review]
Look into https://laravel.com/docs/5.1/authentication#social-authentication for that added fanciness!

**Comment by okapifan** [comment]
I don't have a specific reason to use the u modifier (Do you recommend a other?). And where do I need the more fanciness for?

**Comment by mbicknese** [comment]
Given regex would operate the same without the `u` modifier. So I suppose you can leave it out. The regex is wrong though, have a look at https://regex101.com/r/Vv1RRB/1/tests

Regarding the added fanciness: for starters, do it in another feature / pull request. No need to add extra complexity to this one. See it more as a feature request for later, I love it when I can just press "login with Google" and everything is handled for me. Seeing we use Google Apps we can use the login with Google for our DevMob accounts.

File: app/Http/Controllers/Auth/RegisterController.php [hunk 1]
```diff
@@ -68,6 +71,36 @@ protected function create(array $data)
             'last_name' => $data['last_name'],
             'email' => $data['email'],
             'password' => bcrypt($data['password']),
+            'email_token' => base64_encode($data['email'])
```
Comment (TijmenWierenga) [inline app/Http/Controllers/Auth/RegisterController.php]: This is a security flaw. If the algorithm is exposed it's very easy to create tokens to activate other user's accounts. https://3v4l.org/kGcO0 shows you how this works. It's always the same result. So I could activate your account by just adding the base64 encoded value to the link.

Better to use `str_random` to have a random string which cannot be guessed.

File: app/Http/Controllers/Auth/RegisterController.php [hunk 2]
```diff
@@ -68,6 +71,36 @@ protected function create(array $data)
...
+
+    /**
+     * Handle a registration request for the application.
+     *
+     * @param $token
+     * @return \Illuminate\Http\Response
+     */
+    public function verify($token)
+    {
+        $user = User::where('email_token',$token)->first();
```
Comment (TijmenWierenga) [inline app/Http/Controllers/Auth/RegisterController.php]: Always add a comma + space between arguments: `User::where('email_token, $token)->first();`

File: app/Http/Controllers/Auth/RegisterController.php [hunk 3]
```diff
@@ -68,6 +71,36 @@ protected function create(array $data)
...
+        $user->active = 1;
+        if($user->save()){
```
Comment (TijmenWierenga) [inline app/Http/Controllers/Auth/RegisterController.php]: Always add a space after an `if` statement: `if ($user->save()) { ... }`. This are the rules of PSR2, which is good practise to follow. Google it if you don't know what it is :)

File: app/Http/Controllers/Auth/RegisterController.php (hunk 2 above)
Comment (TijmenWierenga) [reply app/Http/Controllers/Auth/RegisterController.php]: You also have to take care of what happens when the user cannot be found. Maybe the link is broken for some reason. In that case, `$user` will be `null`. Best to handle those situations as well and return a nice error message to the user.

File: app/Jobs/SendVerificationEmail.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Mail;
+use App\Mail\EmailVerification;
+
+class SendVerificationEmail implements ShouldQueue
```
Comment (TijmenWierenga) [inline app/Jobs/SendVerificationEmail.php:13]: Super gedaan! Mooie abstractie

File: app/Http/Controllers/Auth/RegisterController.php [hunk 5]
```diff
@@ -68,6 +71,42 @@ protected function create(array $data)
...
+    public function register(Request $request)
+    {
+        $this->validator($request->all())->validate();
+        event(new Registered($user = $this->create($request->all())));
+        dispatch(new SendVerificationEmail($user));
+        return view('email.verification');
+    }
+
+    /**
+     * Handle a registration request for the application.
```
Comment (mbicknese) [inline app/Http/Controllers/Auth/RegisterController.php]: Just noticed you made a little copy paste mistake here. And no type hint for $token.

File: app/Mail/EmailVerification.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Mail;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Mail\Mailable;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Contracts\Queue\ShouldQueue;
```
Comment (mbicknese) [inline app/Mail/EmailVerification.php]: ShouldQueue imported but not used.

File: app/User.php [hunk 7]
```diff
@@ -15,7 +15,7 @@ class User extends Authenticatable
      * @var array
      */
     protected $fillable = [
-        'first_name', 'last_name', 'email', 'password',
+        'first_name', 'last_name', 'email', 'password', 'email_token',
```
Comment (mbicknese) [inline app/User.php:18]: I'm not enough into Laravel here, but I don't think the email_token should be fillable.

**Review by mbicknese — CHANGES_REQUESTED** [review]

**Comment by mbicknese** [comment]
Except the email_token it's all constructive criticism. The setup looks clean, I think you're using Laravel to its full potential! Going to do a checkout now to see the product in action.

File: app/User.php (hunk 7 above)
Comment (okapifan) [reply app/User.php:18]: It has to be there because when it isn't there it can't generated the token in the create function (RegisterController)

**Review by TijmenWierenga — APPROVED** [review]

**Pull request merged**